### PR TITLE
Document sync rules for Jira and Confluence

### PR DESCRIPTION
**Pull request by praveen-elastic**
## Closes https://github.com/elastic/connectors-py/issues/979, #980 

This PR includes examples for JIra and Confluence Advanced Sync Rules using JQL and CQL queries respectively.

#### Pre-Review Checklist
- [x] this PR has a meaningful title
- [x] this PR links to all relevant github issues that it fixes or partially addresses
- [x] if there is no GH issue, please create it. Each PR should have a link to an issue
- [x] this PR has a thorough description
- [x] Added a label for each target release version (example: `v7.13.2`, `v7.14.0`, `v8.0.0`)
- [x] Considered corresponding documentation changes

### CODE REVIEW

**Review by timgrein — DISMISSED** [review]
LGTM

**Comment by danajuratoni** [comment]
@timgrein is there anything related to validations that is worth mentioning? How are errors presented to the user?
Rest LGTM

**Comment by timgrein** [comment]
@danajuratoni we could a paragraph that errors will be presented to the user visually, yes.

File: docs/reference/confluence.md [hunk 1]
```diff
@@ -92,7 +92,16 @@ The connector syncs the following Confluence object types:
...
 - Permissions are not synced. **All documents** indexed to an Elastic deployment will be visible to **all users with access** to that Elastic Deployment.
-- Filtering rules are not available in the present version. Currently filtering is controlled via ingest pipelines.
+
+## Advanced Sync Rules
+
+Advanced Sync Rules are now available in elastic 8.9 version.
+
+- Users can add [CQL queries](https://developer.atlassian.com/cloud/confluence/advanced-searching-using-cql/) for syncing the source.
+- All the data returned by the CQL queries will be indexed.
+- The valid format for adding sync rules can be referred from [CONFLUENCE.md](../connectors/docs/sync-rules/CONFLUENCE.md). Any errors encountered, will be presented to the user visually in the sync rules overview.
```
Comment (timgrein) [inline docs/reference/confluence.md:102]: You can refer to [CONFLUENCE.md](../connectors/docs/sync-rules/CONFLUENCE.md) for the correct format to add sync rules.

**Review by timgrein — APPROVED** [review]
Added one small suggestion

**Review by timgrein — APPROVED** [review]
LGTM, thank you for addressing the suggestions!

**Pull request merged**